client/history/metadata.go

```diff
@@ -51,66 +51,63 @@ func EncodeClusterShardMD(
 ) (_ ClusterShardID, _ ClusterShardID, _ error) {
-	var sourceClusterShardID ClusterShardID
-	var targetClusterShardID ClusterShardID
+	var clientClusterShardID ClusterShardID
+	var serverClusterShardID ClusterShardID
 
-	clusterNames := clusterShardMD.Get(MetadataKeySourceClusterName)
-	if len(clusterNames) != 1 {
-		return sourceClusterShardID, targetClusterShardID, serviceerror.NewInvalidArgument(fmt.Sprintf(
-			"unable to parse source cluster shard ID: %v",
-			clusterShardMD,
-		))
+	clientClusterID, err := parseInt32(clusterShardMD, MetadataKeyClientClusterID)
+	if err != nil {
+		return clientClusterShardID, serverClusterShardID, err
 	}
-	sourceClusterShardID.ClusterName = clusterNames[0]
-	shardIDs := clusterShardMD.Get(MetadataKeySourceShardID)
-	if len(shardIDs) != 1 {
-		return sourceClusterShardID, targetClusterShardID, serviceerror.NewInvalidArgument(fmt.Sprintf(
-			"unable to parse source cluster shard ID: %v",
-			clusterShardMD,
-		))
+	clientShardID, err := parseInt32(clusterShardMD, MetadataKeyClientShardID)
+	if err != nil {
+		return clientClusterShardID, serverClusterShardID, err
 	}
-	shardID, err := strconv.Atoi(shardIDs[0])
+	clientClusterShardID.ClusterID = clientClusterID
+	clientClusterShardID.ShardID = clientShardID
+
+	serverClusterID, err := parseInt32(clusterShardMD, MetadataKeyServerClusterID)
 	if err != nil {
-		return sourceClusterShardID, targetClusterShardID, serviceerror.NewInvalidArgument(fmt.Sprintf(
-			"unable to parse source cluster shard ID: %v, err: %v",
-			clusterShardMD,
-			err.Error(),
-		))
+		return clientClusterShardID, serverClusterShardID, err
 	}
-	sourceClusterShardID.ShardID = int32(shardID)
-	clusterNames = clusterShardMD.Get(MetadataKeyTargetClusterName)
-	if len(clusterNames) != 1 {
-		return sourceClusterShardID, targetClusterShardID, serviceerror.NewInvalidArgument(fmt.Sprintf(
-			"unable to parse target cluster shard ID: %v",
-			clusterShardMD,
-		))
+	serverShardID, err := parseInt32(clusterShardMD, MetadataKeyServerShardID)
+	if err != nil {
+		return clientClusterShardID, serverClusterShardID, err
 	}
-	targetClusterShardID.ClusterName = clusterNames[0]
-	shardIDs = clusterShardMD.Get(MetadataKeyTargetShardID)
-	if len(shardIDs) != 1 {
-		return sourceClusterShardID, targetClusterShardID, serviceerror.NewInvalidArgument(fmt.Sprintf(
-			"unable to parse source cluster shard ID: %v",
-			clusterShardMD,
+	serverClusterShardID.ClusterID = serverClusterID
+	serverClusterShardID.ShardID = serverShardID
+
+	return clientClusterShardID, serverClusterShardID, nil
+}
+
+func parseInt32(
+	clusterShardMD metadata.MD,
+	metadataKey string,
+) (int32, error) {
+	metadataValues := clusterShardMD.Get(metadataKey)
+	if len(metadataValues) != 1 {
+		return 0, serviceerror.NewInvalidArgument(fmt.Sprintf(
+			"unable to parse metadata key %v: %v",
+			metadataKey,
+			metadataValues,
 		))
 	}
-	shardID, err = strconv.Atoi(shardIDs[0])
+	metadataValue, err := strconv.Atoi(metadataValues[0])
```

Comment:
use `strconv.ParseInt(.., 10, 32)` instead of casting to int32 explicitly.

Reply:
according to the doc, "Atoi is equivalent to ParseInt(s, 10, 0)" so what is the difference? 

strconv.ParseInt(.., 10, 32) will return int64
strconv.Atoi will return int